Are Android dynamic feature modules supported

Hi, does Kover support Android Dynamic Feature Modules? When I add the plugin to a DFM in my multiproject and try to run `feature:dynamicFeature:koverHtmlReportDebug`, I get an error:
```
FAILURE: Build failed with an exception.

* What went wrong:
Task 'koverHtmlReportDebug' not found in project ':feature:dynamicFeature'.
```

If I try to create a multiproject report in my app module, by using `kover(project(":feature:dynamicFeature"))` and running `:app:koverHtmlReportDebug`, I get below error:
```
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':app:koverHtmlReportDebug'.
> Could not resolve all task dependencies for configuration ':app:koverExternalArtifactsDebug'.
   > Could not resolve project :feature:dynamicFeature.
     Required by:
         project :app
      > The consumer was configured to find attribute 'kotlinx.kover.android.build-type' with value 'debug'. However we cannot choose between the following variants of project :feature:dynamicFeature:
          - benchmarkApiElements
          - benchmarkReverseMetadataElements
          - benchmarkRuntimeElements
          - debugApiElements
          - debugReverseMetadataElements
          - debugRuntimeElements
          - koverArtifact
          - releaseApiElements
          - releaseReverseMetadataElements
          - releaseRuntimeElements
        All of them match the consumer attributes:
          - Variant 'benchmarkApiElements' capability myProject.feature:dynamicFeature:unspecified:
              - Unmatched attributes:
                  - Provides com.android.build.api.attributes.AgpVersionAttr '7.3.1' but the consumer didn't ask for it
                  - Provides com.android.build.api.attributes.BuildTypeAttr 'benchmark' but the consumer didn't ask for it
                  - Provides com.android.build.gradle.internal.attributes.VariantAttr 'benchmark' but the consumer didn't ask for it
                  - Doesn't say anything about kotlinx.kover.android.build-type (required 'debug')
                  - Provides org.gradle.category 'library' but the consumer didn't ask for it
                  - Provides org.gradle.jvm.environment 'android' but the consumer didn't ask for it
                  - Provides org.gradle.usage 'java-api' but the consumer didn't ask for it
                  - Provides org.jetbrains.kotlin.platform.type 'androidJvm' but the consumer didn't ask for it
          - Variant 'benchmarkReverseMetadataElements' capability myProject.feature:dynamicFeature:unspecified:
              - Unmatched attributes:
                  - Provides com.android.build.api.attributes.BuildTypeAttr 'benchmark' but the consumer didn't ask for it
                  - Provides com.android.build.gradle.internal.attributes.VariantAttr 'benchmark' but the consumer didn't ask for it
                  - Doesn't say anything about kotlinx.kover.android.build-type (required 'debug')
                  - Provides org.gradle.usage 'android-reverse-meta-data' but the consumer didn't ask for it
          - Variant 'benchmarkRuntimeElements' capability myProject.feature:dynamicFeature:unspecified:
              - Unmatched attributes:
                  - Provides com.android.build.api.attributes.AgpVersionAttr '7.3.1' but the consumer didn't ask for it
                  - Provides com.android.build.api.attributes.BuildTypeAttr 'benchmark' but the consumer didn't ask for it
                  - Provides com.android.build.gradle.internal.attributes.VariantAttr 'benchmark' but the consumer didn't ask for it
                  - Doesn't say anything about kotlinx.kover.android.build-type (required 'debug')
                  - Provides org.gradle.category 'library' but the consumer didn't ask for it
                  - Provides org.gradle.jvm.environment 'android' but the consumer didn't ask for it
                  - Provides org.gradle.usage 'java-runtime' but the consumer didn't ask for it
                  - Provides org.jetbrains.kotlin.platform.type 'androidJvm' but the consumer didn't ask for it
          - Variant 'debugApiElements' capability myProject.feature:dynamicFeature:unspecified:
              - Unmatched attributes:
                  - Provides com.android.build.api.attributes.AgpVersionAttr '7.3.1' but the consumer didn't ask for it
                  - Provides com.android.build.api.attributes.BuildTypeAttr 'debug' but the consumer didn't ask for it
                  - Provides com.android.build.gradle.internal.attributes.VariantAttr 'debug' but the consumer didn't ask for it
                  - Doesn't say anything about kotlinx.kover.android.build-type (required 'debug')
                  - Provides org.gradle.category 'library' but the consumer didn't ask for it
                  - Provides org.gradle.jvm.environment 'android' but the consumer didn't ask for it
                  - Provides org.gradle.usage 'java-api' but the consumer didn't ask for it
                  - Provides org.jetbrains.kotlin.platform.type 'androidJvm' but the consumer didn't ask for it
          - Variant 'debugReverseMetadataElements' capability myProject.feature:dynamicFeature:unspecified:
              - Unmatched attributes:
                  - Provides com.android.build.api.attributes.BuildTypeAttr 'debug' but the consumer didn't ask for it
                  - Provides com.android.build.gradle.internal.attributes.VariantAttr 'debug' but the consumer didn't ask for it
                  - Doesn't say anything about kotlinx.kover.android.build-type (required 'debug')
                  - Provides org.gradle.usage 'android-reverse-meta-data' but the consumer didn't ask for it
          - Variant 'debugRuntimeElements' capability myProject.feature:dynamicFeature:unspecified:
              - Unmatched attributes:
                  - Provides com.android.build.api.attributes.AgpVersionAttr '7.3.1' but the consumer didn't ask for it
                  - Provides com.android.build.api.attributes.BuildTypeAttr 'debug' but the consumer didn't ask for it
                  - Provides com.android.build.gradle.internal.attributes.VariantAttr 'debug' but the consumer didn't ask for it
                  - Doesn't say anything about kotlinx.kover.android.build-type (required 'debug')
                  - Provides org.gradle.category 'library' but the consumer didn't ask for it
                  - Provides org.gradle.jvm.environment 'android' but the consumer didn't ask for it
                  - Provides org.gradle.usage 'java-runtime' but the consumer didn't ask for it
                  - Provides org.jetbrains.kotlin.platform.type 'androidJvm' but the consumer didn't ask for it
          - Variant 'koverArtifact' capability myProject.feature:dynamicFeature:unspecified:
              - Unmatched attributes:
                  - Doesn't say anything about kotlinx.kover.android.build-type (required 'debug')
                  - Provides kotlinx.kover.project.path ':feature:dynamicFeature' but the consumer didn't ask for it
                  - Provides kotlinx.kover.variant.name '' but the consumer didn't ask for it
          - Variant 'releaseApiElements' capability myProject.feature:dynamicFeature:unspecified:
              - Unmatched attributes:
                  - Provides com.android.build.api.attributes.AgpVersionAttr '7.3.1' but the consumer didn't ask for it
                  - Provides com.android.build.api.attributes.BuildTypeAttr 'release' but the consumer didn't ask for it
                  - Provides com.android.build.gradle.internal.attributes.VariantAttr 'release' but the consumer didn't ask for it
                  - Doesn't say anything about kotlinx.kover.android.build-type (required 'debug')
                  - Provides org.gradle.category 'library' but the consumer didn't ask for it
                  - Provides org.gradle.jvm.environment 'android' but the consumer didn't ask for it
                  - Provides org.gradle.usage 'java-api' but the consumer didn't ask for it
                  - Provides org.jetbrains.kotlin.platform.type 'androidJvm' but the consumer didn't ask for it
          - Variant 'releaseReverseMetadataElements' capability myProject.feature:dynamicFeature:unspecified:
              - Unmatched attributes:
                  - Provides com.android.build.api.attributes.BuildTypeAttr 'release' but the consumer didn't ask for it
                  - Provides com.android.build.gradle.internal.attributes.VariantAttr 'release' but the consumer didn't ask for it
                  - Doesn't say anything about kotlinx.kover.android.build-type (required 'debug')
                  - Provides org.gradle.usage 'android-reverse-meta-data' but the consumer didn't ask for it
          - Variant 'releaseRuntimeElements' capability myProject.feature:dynamicFeature:unspecified:
              - Unmatched attributes:
                  - Provides com.android.build.api.attributes.AgpVersionAttr '7.3.1' but the consumer didn't ask for it
                  - Provides com.android.build.api.attributes.BuildTypeAttr 'release' but the consumer didn't ask for it
                  - Provides com.android.build.gradle.internal.attributes.VariantAttr 'release' but the consumer didn't ask for it
                  - Doesn't say anything about kotlinx.kover.android.build-type (required 'debug')
                  - Provides org.gradle.category 'library' but the consumer didn't ask for it
                  - Provides org.gradle.jvm.environment 'android' but the consumer didn't ask for it
                  - Provides org.gradle.usage 'java-runtime' but the consumer didn't ask for it
                  - Provides org.jetbrains.kotlin.platform.type 'androidJvm' but the consumer didn't ask for it
```

The plugin works as expected for all non-DFM modules in my project.

--- a/kover-gradle-plugin/examples/android/dynamic/app/build.gradle.kts
+++ b/kover-gradle-plugin/examples/android/dynamic/app/build.gradle.kts
@@ -33,6 +33,7 @@ android {
     buildFeatures {
         viewBinding = true
     }
+     dynamicFeatures = [":dyn"]
 }
 
 dependencies {

--- a/kover-gradle-plugin/examples/android/dynamic/app/build.gradle.kts
+++ b/kover-gradle-plugin/examples/android/dynamic/app/build.gradle.kts
@@ -33,7 +33,7 @@ android {
     buildFeatures {
         viewBinding = true
     }
-     dynamicFeatures = [":dyn"]
+     dynamicFeatures += ":dyn"
 }
 
 dependencies {

--- a/kover-gradle-plugin/examples/android/dynamic/app/build.gradle.kts
+++ b/kover-gradle-plugin/examples/android/dynamic/app/build.gradle.kts
@@ -1,0 +1,44 @@
+plugins {
+    id ("com.android.application")
+    id ("org.jetbrains.kotlin.android")
+    id ("org.jetbrains.kotlinx.kover")
+}
+
+android {
+    namespace = "kotlinx.kover.test.android"
+    compileSdk = 32
+
+    defaultConfig {
+        applicationId = "kotlinx.kover.test.android"
+        minSdk = 21
+        targetSdk = 31
+        versionCode = 1
+        versionName = "1.0"
+
+        testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+    }
+
+    buildTypes {
+        release {
+            isMinifyEnabled = true
+        }
+    }
+    compileOptions {
+        sourceCompatibility = JavaVersion.VERSION_1_8
+        targetCompatibility = JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = "1.8"
+    }
+    buildFeatures {
+        viewBinding = true
+    }
+}
+
+dependencies {
+    implementation("androidx.core:core-ktx:1.8.0")
+    implementation("androidx.appcompat:appcompat:1.5.0")
+    implementation("com.google.android.material:material:1.6.1")
+    implementation("androidx.constraintlayout:constraintlayout:2.1.4")
+    testImplementation("junit:junit:4.13.2")
+}

--- a/kover-gradle-plugin/examples/android/dynamic/app/src/main/AndroidManifest.xml
+++ b/kover-gradle-plugin/examples/android/dynamic/app/src/main/AndroidManifest.xml
@@ -1,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android">
+
+    <application android:label="@string/app_name">
+        <uses-library android:name="com.google.android.things" android:required="false" />
+
+        <activity android:name=".MainActivity"
+                  android:exported="true">
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN" />
+
+                <category android:name="android.intent.category.LAUNCHER" />
+            </intent-filter>
+            <!-- Make this the first activity that is displayed when the device boots. -->
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN" />
+
+                <category android:name="android.intent.category.HOME" />
+                <category android:name="android.intent.category.DEFAULT" />
+            </intent-filter>
+        </activity>
+    </application>
+
+</manifest>

--- a/kover-gradle-plugin/examples/android/dynamic/app/src/main/java/kotlinx/kover/test/android/DebugUtil.kt
+++ b/kover-gradle-plugin/examples/android/dynamic/app/src/main/java/kotlinx/kover/test/android/DebugUtil.kt
@@ -1,0 +1,7 @@
+package kotlinx.kover.test.android
+
+object DebugUtil {
+    fun log(message: String) {
+        println("DEBUG: $message")
+    }
+}

--- a/kover-gradle-plugin/examples/android/dynamic/app/src/main/java/kotlinx/kover/test/android/MainActivity.kt
+++ b/kover-gradle-plugin/examples/android/dynamic/app/src/main/java/kotlinx/kover/test/android/MainActivity.kt
@@ -1,0 +1,13 @@
+package kotlinx.kover.test.android
+
+import android.os.Bundle
+import android.app.Activity
+
+class MainActivity : Activity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+    }
+
+}

--- a/kover-gradle-plugin/examples/android/dynamic/app/src/main/java/kotlinx/kover/test/android/Maths.kt
+++ b/kover-gradle-plugin/examples/android/dynamic/app/src/main/java/kotlinx/kover/test/android/Maths.kt
@@ -1,0 +1,13 @@
+package kotlinx.kover.test.android
+
+object Maths {
+    fun sum(a: Int, b: Int): Int {
+        DebugUtil.log("invoked sum")
+        return a + b
+    }
+
+    fun sub(a: Int, b: Int): Int {
+        DebugUtil.log("invoked sub")
+        return a - b
+    }
+}

--- a/kover-gradle-plugin/examples/android/dynamic/app/src/main/res/layout/activity_main.xml
+++ b/kover-gradle-plugin/examples/android/dynamic/app/src/main/res/layout/activity_main.xml
@@ -1,0 +1,36 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".MainActivity">
+
+    <TextView
+        android:id="@+id/main_label"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="104dp"
+        android:layout_marginTop="28dp"
+        android:text="SERIALIZATION TEST"
+        android:textSize="20sp"
+        android:textStyle="bold"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <EditText
+        android:id="@+id/encoded_text"
+        android:layout_width="373dp"
+        android:layout_height="411dp"
+        android:layout_marginStart="16dp"
+        android:layout_marginTop="16dp"
+        android:editable="false"
+        android:ems="10"
+        android:gravity="start|top"
+        android:inputType="textMultiLine"
+        android:textAlignment="viewStart"
+        android:textSize="12sp"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/main_label" />
+
+</androidx.constraintlayout.widget.ConstraintLayout>

--- a/kover-gradle-plugin/examples/android/dynamic/app/src/main/res/values/colors.xml
+++ b/kover-gradle-plugin/examples/android/dynamic/app/src/main/res/values/colors.xml
@@ -1,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <color name="purple_200">#FFBB86FC</color>
+    <color name="purple_500">#FF6200EE</color>
+    <color name="purple_700">#FF3700B3</color>
+    <color name="teal_200">#FF03DAC5</color>
+    <color name="teal_700">#FF018786</color>
+    <color name="black">#FF000000</color>
+    <color name="white">#FFFFFFFF</color>
+</resources>

--- a/kover-gradle-plugin/examples/android/dynamic/app/src/main/res/values/strings.xml
+++ b/kover-gradle-plugin/examples/android/dynamic/app/src/main/res/values/strings.xml
@@ -1,0 +1,3 @@
+<resources>
+    <string name="app_name">Android Test</string>
+</resources>

--- a/kover-gradle-plugin/examples/android/dynamic/app/src/main/res/values/themes.xml
+++ b/kover-gradle-plugin/examples/android/dynamic/app/src/main/res/values/themes.xml
@@ -1,0 +1,4 @@
+<resources>
+
+    <style name="Theme.App" parent="android:Theme.Material.Light.DarkActionBar" />
+</resources>

--- a/kover-gradle-plugin/examples/android/dynamic/app/src/test/java/kotlinx/kover/test/android/LocalTests.kt
+++ b/kover-gradle-plugin/examples/android/dynamic/app/src/test/java/kotlinx/kover/test/android/LocalTests.kt
@@ -1,0 +1,13 @@
+package kotlinx.kover.test.android
+
+import org.junit.Test
+
+import org.junit.Assert.*
+
+
+class LocalTests {
+    @Test
+    fun testDebugUtils() {
+        assertEquals(3, Maths.sum(1, 2))
+    }
+}

--- a/kover-gradle-plugin/examples/android/dynamic/build.gradle.kts
+++ b/kover-gradle-plugin/examples/android/dynamic/build.gradle.kts
@@ -1,0 +1,6 @@
+plugins {
+    id("com.android.application") version "7.4.0" apply false
+    id("com.android.library") version "7.4.0" apply false
+    id("org.jetbrains.kotlin.android") version "1.8.20" apply false
+    id("org.jetbrains.kotlinx.kover") version "0.7.3" apply false
+}

--- a/kover-gradle-plugin/examples/android/dynamic/build.gradle.kts
+++ b/kover-gradle-plugin/examples/android/dynamic/build.gradle.kts
@@ -1,6 +1,7 @@
 plugins {
     id("com.android.application") version "7.4.0" apply false
     id("com.android.library") version "7.4.0" apply false
+    id ("com.android.dynamic-feature") version "7.4.0" apply false
     id("org.jetbrains.kotlin.android") version "1.8.20" apply false
     id("org.jetbrains.kotlinx.kover") version "0.7.3" apply false
 }

--- a/kover-gradle-plugin/examples/android/dynamic/dyn/build.gradle.kts
+++ b/kover-gradle-plugin/examples/android/dynamic/dyn/build.gradle.kts
@@ -1,5 +1,5 @@
 plugins {
-    id ("com.android.library")
+    id ("com.android.dynamic-feature")
     id ("org.jetbrains.kotlin.android")
     id ("org.jetbrains.kotlinx.kover")
 }
@@ -15,11 +15,6 @@ android {
         testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
     }
 
-    buildTypes {
-        release {
-            isMinifyEnabled = true
-        }
-    }
     compileOptions {
         sourceCompatibility = JavaVersion.VERSION_1_8
         targetCompatibility = JavaVersion.VERSION_1_8

--- a/kover-gradle-plugin/examples/android/dynamic/dyn/build.gradle.kts
+++ b/kover-gradle-plugin/examples/android/dynamic/dyn/build.gradle.kts
@@ -1,0 +1,82 @@
+plugins {
+    id ("com.android.library")
+    id ("org.jetbrains.kotlin.android")
+    id ("org.jetbrains.kotlinx.kover")
+}
+
+android {
+    namespace = "kotlinx.kover.test.android"
+
+    compileSdk = 32
+
+    defaultConfig {
+        minSdk = 21
+
+        testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+    }
+
+    buildTypes {
+        release {
+            isMinifyEnabled = true
+        }
+    }
+    compileOptions {
+        sourceCompatibility = JavaVersion.VERSION_1_8
+        targetCompatibility = JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = "1.8"
+    }
+}
+
+dependencies {
+    implementation("androidx.core:core-ktx:1.8.0")
+    implementation("androidx.appcompat:appcompat:1.5.0")
+    testImplementation("junit:junit:4.13.2")
+}
+
+
+/*
+ * Kover configs
+ */
+
+dependencies {
+    kover(project(":app"))
+}
+
+
+koverReport {
+    // filters for all report types of all build variants
+    filters {
+        excludes {
+            classes(
+                "*Fragment",
+                "*Fragment\$*",
+                "*Activity",
+                "*Activity\$*",
+                "*.databinding.*",
+                "*.BuildConfig"
+            )
+        }
+    }
+
+    androidReports("release") {
+        // filters for all report types only of 'release' build type
+        filters {
+            excludes {
+                classes(
+                    "*Fragment",
+                    "*Fragment\$*",
+                    "*Activity",
+                    "*Activity\$*",
+                    "*.databinding.*",
+                    "*.BuildConfig",
+
+                    // excludes debug classes
+                    "*.DebugUtil"
+                )
+            }
+        }
+    }
+}
+

--- a/kover-gradle-plugin/examples/android/dynamic/dyn/build.gradle.kts
+++ b/kover-gradle-plugin/examples/android/dynamic/dyn/build.gradle.kts
@@ -33,6 +33,7 @@ dependencies {
     implementation("androidx.core:core-ktx:1.8.0")
     implementation("androidx.appcompat:appcompat:1.5.0")
     testImplementation("junit:junit:4.13.2")
+    implementation(project(":app"))
 }
 
 

--- a/kover-gradle-plugin/examples/android/dynamic/dyn/src/main/AndroidManifest.xml
+++ b/kover-gradle-plugin/examples/android/dynamic/dyn/src/main/AndroidManifest.xml
@@ -1,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools">
+
+    <application
+        android:allowBackup="true"
+        android:supportsRtl="true"
+        tools:targetApi="31">
+    </application>
+
+</manifest>

--- a/kover-gradle-plugin/examples/android/dynamic/dyn/src/main/AndroidManifest.xml
+++ b/kover-gradle-plugin/examples/android/dynamic/dyn/src/main/AndroidManifest.xml
@@ -1,11 +1,21 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:tools="http://schemas.android.com/tools">
+    xmlns:tools="http://schemas.android.com/tools"
+          xmlns:dist="http://schemas.android.com/apk/distribution">
 
     <application
         android:allowBackup="true"
         android:supportsRtl="true"
         tools:targetApi="31">
     </application>
+
+    <dist:module
+            dist:instant="false"
+            dist:title="Dynamic title">
+        <dist:delivery>
+            <dist:install-time />
+        </dist:delivery>
+        <dist:fusing dist:include="false" />
+    </dist:module>
 
 </manifest>

--- a/kover-gradle-plugin/examples/android/dynamic/dyn/src/main/java/kotlinx/kover/test/android/dyn/DebugUtil.kt
+++ b/kover-gradle-plugin/examples/android/dynamic/dyn/src/main/java/kotlinx/kover/test/android/dyn/DebugUtil.kt
@@ -1,0 +1,7 @@
+package kotlinx.kover.test.android.dyn
+
+object DebugUtil {
+    fun log(message: String) {
+        println("LIB DEBUG: $message")
+    }
+}

--- a/kover-gradle-plugin/examples/android/dynamic/dyn/src/main/java/kotlinx/kover/test/android/dyn/MagicFactory.kt
+++ b/kover-gradle-plugin/examples/android/dynamic/dyn/src/main/java/kotlinx/kover/test/android/dyn/MagicFactory.kt
@@ -1,0 +1,8 @@
+package kotlinx.kover.test.android.dyn
+
+object MagicFactory {
+    fun generate(): Int {
+        DebugUtil.log("generate Int")
+        return 42
+    }
+}

--- a/kover-gradle-plugin/examples/android/dynamic/dyn/src/test/java/kotlinx/kover/test/android/LocalLibTests.kt
+++ b/kover-gradle-plugin/examples/android/dynamic/dyn/src/test/java/kotlinx/kover/test/android/LocalLibTests.kt
@@ -1,0 +1,14 @@
+package kotlinx.kover.test.android
+
+import kotlinx.kover.test.android.dyn.MagicFactory
+import org.junit.Test
+
+import org.junit.Assert.*
+
+
+class LocalLibTests {
+    @Test
+    fun testDebugUtils() {
+        assertEquals(42, MagicFactory.generate())
+    }
+}

--- a/kover-gradle-plugin/examples/android/dynamic/gradle.properties
+++ b/kover-gradle-plugin/examples/android/dynamic/gradle.properties
@@ -1,0 +1,23 @@
+# Project-wide Gradle settings.
+# IDE (e.g. Android Studio) users:
+# Gradle settings configured through the IDE *will override*
+# any settings specified in this file.
+# For more details on how to configure your build environment visit
+# http://www.gradle.org/docs/current/userguide/build_environment.html
+# Specifies the JVM arguments used for the daemon process.
+# The setting is particularly useful for tweaking memory settings.
+org.gradle.jvmargs=-Xmx2048m -Dfile.encoding=UTF-8
+# When configured, Gradle will run in incubating parallel mode.
+# This option should only be used with decoupled projects. More details, visit
+# http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
+# org.gradle.parallel=true
+# AndroidX package structure to make it clearer which packages are bundled with the
+# Android operating system, and which are packaged with your app's APK
+# https://developer.android.com/topic/libraries/support-library/androidx-rn
+android.useAndroidX=true
+# Kotlin code style for this project: "official" or "obsolete":
+kotlin.code.style=official
+# Enables namespacing of each library's R class so that its R class includes only the
+# resources declared in the library itself and none from the library's dependencies,
+# thereby reducing the size of the R class for that library
+android.nonTransitiveRClass=true

--- a/kover-gradle-plugin/examples/android/dynamic/settings.gradle.kts
+++ b/kover-gradle-plugin/examples/android/dynamic/settings.gradle.kts
@@ -1,0 +1,19 @@
+pluginManagement {
+    repositories {
+        google()
+        mavenCentral()
+        gradlePluginPortal()
+    }
+}
+
+dependencyResolutionManagement {
+    repositoriesMode.set(RepositoriesMode.FAIL_ON_PROJECT_REPOS)
+    repositories {
+        google()
+        mavenCentral()
+        gradlePluginPortal()
+    }
+}
+
+rootProject.name = "android_dynamic"
+include(":app", ":dyn")

--- a/kover-gradle-plugin/src/functionalTest/kotlin/kotlinx/kover/gradle/plugin/test/functional/cases/DynamicFeatureTests.kt
+++ b/kover-gradle-plugin/src/functionalTest/kotlin/kotlinx/kover/gradle/plugin/test/functional/cases/DynamicFeatureTests.kt
@@ -1,0 +1,17 @@
+package kotlinx.kover.gradle.plugin.test.functional.cases
+
+import kotlinx.kover.gradle.plugin.test.functional.framework.checker.CheckerContext
+import kotlinx.kover.gradle.plugin.test.functional.framework.starter.ExamplesTest
+
+internal class DynamicFeatureTests {
+    @ExamplesTest("android/dynamic", commands = [":dyn:koverXmlReportRelease"])
+    fun CheckerContext.test() {
+        subproject("dyn") {
+            xmlReport("release") {
+                classCounter("kotlinx.kover.test.android.DebugUtil").assertAbsent()
+                methodCounter("kotlinx.kover.test.android.Maths", "sum").assertCovered()
+                methodCounter("kotlinx.kover.test.android.dyn.MagicFactory", "generate").assertCovered()
+            }
+        }
+    }
+}

--- a/kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/locators/CompilationsListeners.kt
+++ b/kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/locators/CompilationsListeners.kt
@@ -130,3 +130,4 @@ internal const val ANDROID_APP_PLUGIN_ID = "com.android.application"
 
 internal const val ANDROID_LIB_PLUGIN_ID = "com.android.library"
 
+internal const val ANDROID_DYNAMIC_PLUGIN_ID = "com.android.dynamic-feature"

--- a/kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/locators/KotlinAndroidLocator.kt
+++ b/kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/locators/KotlinAndroidLocator.kt
@@ -24,6 +24,9 @@ internal fun LocatorContext.initKotlinAndroidPluginLocator() {
     project.pluginManager.withPlugin(ANDROID_LIB_PLUGIN_ID) {
         project.afterAndroidPluginApplied { processAndroidTarget() }
     }
+    project.pluginManager.withPlugin(ANDROID_DYNAMIC_PLUGIN_ID) {
+        project.afterAndroidPluginApplied { processAndroidTarget() }
+    }
 }
 
 private fun LocatorContext.processAndroidTarget() {

